quesma/ingest/processor.go

```diff
@@ -982,9 +981,9 @@ func (ip *IngestProcessor) Ping() error {
 	return ip.chDb.Ping()
 }
 
-func NewIngestProcessor(cfg *config.QuesmaConfiguration, chDb quesma_api.BackendConnector, phoneHomeAgent telemetry.PhoneHomeAgent, loader chLib.TableDiscovery, schemaRegistry schema.Registry, virtualTableStorage persistence.JSONDatabase, tableResolver table_resolver.TableResolver) *IngestProcessor {
+func NewIngestProcessor(cfg *config.QuesmaConfiguration, chDb quesma_api.BackendConnector, phoneHomeClient diag.PhoneHomeClient, loader chLib.TableDiscovery, schemaRegistry schema.Registry, virtualTableStorage persistence.JSONDatabase, tableResolver table_resolver.TableResolver) *IngestProcessor {
```

Comment:
Have not checked that by implementing that, however I think we can reduce number of parameters by getting rid of `virtualTableStorage ` and `tableDiscover`. The whole idea is based on the fact that `TableDiscovery ` is a field of `schemaRegistry` and `virtualTableStorage` is a field of `TableDiscovery`, so dependency graph looks as follows:

`virtualTableStorage` <- `tableDiscover` <- `schemaRegistry`

Reply:
As chatted on slack, it'd currently cause larger refactor due to its exposure via `TableDiscoveryTableProviderAdapter` :(